src/index/CompressedRelation.cpp

```diff
@@ -1587,15 +1661,30 @@ auto CompressedRelationWriter::createPermutationPair(
     for (size_t idx : ad_utility::integerRange(block.numRows())) {
       Id col0Id = firstCol[idx];
       decltype(auto) curRemainingCols = permutedCols[idx];
+
       if (col0Id != col0IdCurrentRelation) {
         finishRelation();
         col0IdCurrentRelation = col0Id;
       }
+
+      // Check if we need to create a new block before adding the current
+      // triple. We create a new block if:
+      // 1. The relation buffer is at the block size limit, AND
+      // 2. The current triple has different first three columns than the last
+      //    triple in the buffer (to ensure equal triples stay in same block)
+      if (relation.size() >= blocksize && !relation.empty()) {
```

Comment:
```suggestion
      AD_CORRECTNESS_CHECK(blocksize > 0);
      if (relation.size() >= blocksize) {
```

Since blocksize should be always at least 1, this always means that the relation is not empty.